src/main/java/edu/cornell/kfs/fp/batch/service/impl/AccountingXmlDocumentDownloadAttachmentServiceImpl.java

```diff
@@ -34,34 +35,37 @@
         if (StringUtils.isBlank(accountingXmlDocumentBackupLink.getCredentialGroupCode())) {
             LOG.error("createAttachmentFromBackupLink, the Credential Group Code is blank");
-            throw new IOException("Unable to download attachment with blank Credential Group Code: " + accountingXmlDocumentBackupLink.getLinkUrl());
+            throw new ValidationException("Unable to download attachment with blank Credential Group Code: " + accountingXmlDocumentBackupLink.getLinkUrl());
         }
-        
-        byte[] formFile = downloadByteArray(accountingXmlDocumentBackupLink);
-        
-        if (formFile.length > 0) {
-            String uploadFileName = accountingXmlDocumentBackupLink.getFileName();
-            String mimeType = URLConnection.guessContentTypeFromName(uploadFileName);
-            int fileSize = (int) formFile.length;
-            String attachmentType = null;
-            
-            if (LOG.isDebugEnabled()) {
-                LOG.debug("createAttachmentFromBackupLink, uploadFileName: " + uploadFileName + " mimeType: " + mimeType
-                    + " fileSize: " + fileSize);
+        try {
+            byte[] formFile = downloadByteArray(accountingXmlDocumentBackupLink);
+
+            if (formFile.length > 0) {
+                String uploadFileName = accountingXmlDocumentBackupLink.getFileName();
+                String mimeType = URLConnection.guessContentTypeFromName(uploadFileName);
+                int fileSize = (int) formFile.length;
+                String attachmentType = null;
+
+                if (LOG.isDebugEnabled()) {
+                    LOG.debug("createAttachmentFromBackupLink, uploadFileName: " + uploadFileName + " mimeType: " + mimeType
+                            + " fileSize: " + fileSize);
+                }
+                InputStream inputStream = new ByteArrayInputStream(formFile);
+
+                Attachment attachment = attachmentService.createAttachment(document, uploadFileName, mimeType, fileSize, inputStream, attachmentType);
+                return attachment;
+
+            } else {
+                LOG.error("createAttachmentFromBackupLink, the form file is NULL");
+                throw new ValidationException("The form file is NULL: " + accountingXmlDocumentBackupLink.getLinkUrl());
             }
-            InputStream inputStream = new ByteArrayInputStream(formFile);
-            Attachment attachment = attachmentService.createAttachment(document, uploadFileName, mimeType, fileSize,
-                    inputStream, attachmentType);
-            return attachment;
-        } else {
-            LOG.error("createAttachmentFromBackupLink, the form file is NULL");
-        }
-        
-        throw new IOException("Unable to download attachment: " + accountingXmlDocumentBackupLink.getLinkUrl());
+        } catch (IOException e) {
+            LOG.error("Unable to download attachment: " + accountingXmlDocumentBackupLink.getLinkUrl());
```

Comment:
This logging statement should start with the name of the enclosing method, for consistency with how we've been setting up other such log messages. Also, I think this logging call should pass in the exception as the second argument, so that we can view its stack trace in the logs.

Reply:
Thank you Chad, I will update the code accordingly.